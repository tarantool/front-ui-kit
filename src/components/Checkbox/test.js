import React from 'react';
import renderer from 'react-test-renderer';
import { Checkbox } from './index';

it('Checkbox renders correctly', () => {
  const changeState = () => null;

  const tree = renderer.create(
    <>
      <Checkbox indeterminate onChange={changeState} title='Indeterminate'>Checked</Checkbox>
      <Checkbox checked indeterminate onChange={changeState} >Checked</Checkbox>
      <Checkbox checked onChange={changeState} title='Initial'>Checked</Checkbox>
      <Checkbox onChange={changeState}>Initial</Checkbox>
      <Checkbox disabled checked indeterminate onChange={changeState}>Checked disabled</Checkbox>
      <Checkbox disabled indeterminate onChange={changeState}>Disabled</Checkbox>
      <Checkbox disabled checked onChange={changeState}>Checked disabled</Checkbox>
      <Checkbox disabled onChange={changeState}>Disabled</Checkbox>
    </>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
