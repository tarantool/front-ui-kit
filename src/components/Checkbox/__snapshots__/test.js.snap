// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders correctly 1`] = `
Array [
  <label
    className="css-759u60"
    title="Indeterminate"
  >
    <input
      checked={false}
      className="css-1fb08im"
      disabled={false}
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="css-1ysxw4w"
    >
      <svg
        className="css-1v31moo"
        viewBox="0 0 16 16"
      >
        <use
          xlinkHref="#checkbox-indeterminate"
        />
      </svg>
    </div>
    <span
      className="css-1ggp6x0"
    >
      Checked
    </span>
  </label>,
  <label
    className="css-759u60"
  >
    <input
      checked={true}
      className="css-1fb08im"
      disabled={false}
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="css-1ysxw4w"
    >
      <svg
        className="css-1v31moo"
        viewBox="0 0 16 16"
      >
        <use
          xlinkHref="#checkbox-indeterminate"
        />
      </svg>
    </div>
    <span
      className="css-1ggp6x0"
    >
      Checked
    </span>
  </label>,
  <label
    className="css-759u60"
    title="Initial"
  >
    <input
      checked={true}
      className="css-1fb08im"
      disabled={false}
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="css-1ysxw4w"
    >
      <svg
        className="css-1v31moo"
        viewBox="0 0 16 16"
      >
        <use
          xlinkHref="#checkbox-checked"
        />
      </svg>
    </div>
    <span
      className="css-1ggp6x0"
    >
      Checked
    </span>
  </label>,
  <label
    className="css-759u60"
  >
    <input
      checked={false}
      className="css-1fb08im"
      disabled={false}
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="css-1ysxw4w"
    >
      <svg
        className="css-1v31moo"
        viewBox="0 0 16 16"
      >
        <use
          xlinkHref="#checkbox"
        />
      </svg>
    </div>
    <span
      className="css-1ggp6x0"
    >
      Initial
    </span>
  </label>,
  <label
    className="css-759u60"
  >
    <input
      checked={true}
      className="css-1fb08im"
      disabled={true}
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="css-1ysxw4w"
    >
      <svg
        className="css-17jpq69"
        viewBox="0 0 16 16"
      >
        <use
          xlinkHref="#checkbox-indeterminate-disabled"
        />
      </svg>
    </div>
    <span
      className="css-1ggp6x0"
    >
      Checked disabled
    </span>
  </label>,
  <label
    className="css-759u60"
  >
    <input
      checked={false}
      className="css-1fb08im"
      disabled={true}
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="css-1ysxw4w"
    >
      <svg
        className="css-17jpq69"
        viewBox="0 0 16 16"
      >
        <use
          xlinkHref="#checkbox-indeterminate-disabled"
        />
      </svg>
    </div>
    <span
      className="css-1ggp6x0"
    >
      Disabled
    </span>
  </label>,
  <label
    className="css-759u60"
  >
    <input
      checked={true}
      className="css-1fb08im"
      disabled={true}
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="css-1ysxw4w"
    >
      <svg
        className="css-17jpq69"
        viewBox="0 0 16 16"
      >
        <use
          xlinkHref="#checkbox-checked-disabled"
        />
      </svg>
    </div>
    <span
      className="css-1ggp6x0"
    >
      Checked disabled
    </span>
  </label>,
  <label
    className="css-759u60"
  >
    <input
      checked={false}
      className="css-1fb08im"
      disabled={true}
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="css-1ysxw4w"
    >
      <svg
        className="css-17jpq69"
        viewBox="0 0 16 16"
      >
        <use
          xlinkHref="#checkbox-disabled"
        />
      </svg>
    </div>
    <span
      className="css-1ggp6x0"
    >
      Disabled
    </span>
  </label>,
]
`;
